cairo/ethereum/cancun/vm/interpreter.cairo

```diff
@@ -334,3 +353,160 @@ func _execute_code{
     // Recursive call to continue execution
     return _execute_code(evm);
 }
+
+func process_message_call{
+    range_check_ptr,
+    bitwise_ptr: BitwiseBuiltin*,
+    keccak_ptr: KeccakBuiltin*,
+    poseidon_ptr: PoseidonBuiltin*,
+}(message: Message, env: Environment) -> (MessageCallOutput, EthereumException*) {
+    alloc_locals;
+
+    // Check if this is a contract creation (target is empty)
+    if (cast(message.value.target.value.address, felt) == 0) {
+        let state = env.value.state;
+        let has_collision = account_has_code_or_nonce{state=state}(message.value.current_target);
+        EnvImpl.set_state{env=env}(state);
+        if (has_collision.value != FALSE) {
+            // Return early with collision error
+            tempvar collision_error = new EthereumException(AddressCollision);
+            let msg = create_empty_message_call_output(collision_error);
+            tempvar msg = msg;
+            tempvar err = cast(0, EthereumException*);
+            return (msg, err);
+        }
+
+        // Process create message
+        let (evm, err) = process_create_message(message, env);
+        if (cast(err, felt) != 0) {
+            let msg = create_empty_message_call_output(cast(0, EthereumException*));
+            tempvar msg = msg;
```

Comment:
anytime you return in this function, you need to drop all used dicts from the evm struct. I would create a helper function that's like `drop_evm` which calls `dict_squash` on all inner dicts